### PR TITLE
Filter content-text in marketing boxes

### LINKED ISSUE #34: Filters do not work in marketing spots
All URLs are SSO urls right now, regardless of the currently used domain.

### PR DESCRIPTION
**Pull request by justusdieckmann**
image
Fixes #34.

### CODE REVIEW

File: classes/output/core_renderer.php [hunk 1]
```diff
@@ -960,6 +960,8 @@ private function get_login() {
             return false;
         }
 
+        return ['url' => get_login_url()];
```
Comment (Dagefoerde) [inline classes/output/core_renderer.php:963]: this looks wrong :D is this a leftover from debugging?

File: layout/frontpage.php [hunk 2]
```diff
@@ -36,7 +36,7 @@
     $marketingspot = $i + 1;
     $marketingboxes[$i]->index = $i;
     $marketingboxes[$i]->title = get_config('theme_wwu2019', 'marketing' . $marketingspot);
-    $marketingboxes[$i]->content = get_config('theme_wwu2019', 'marketing' . $marketingspot . 'content');
+    $marketingboxes[$i]->content = format_text(get_config('theme_wwu2019', 'marketing' . $marketingspot . 'content'));
```
Comment (Dagefoerde) [inline layout/frontpage.php:39]: @tobiasreischmann do you recall whether this is the right one for filtering content (not titles)?

File: classes/output/core_renderer.php (hunk 1 above)
Comment (justusdieckmann) [reply classes/output/core_renderer.php]: Ups ^^

File: layout/frontpage.php (hunk 2 above)
Comment (justusdieckmann) [reply layout/frontpage.php:39]: If you're just thinking about `format_text`/`format_string`, the moodle docs say this:

> ### format_string()
> This function should be used to:
> - print short strings (non html) that need filter processing (activity titles, post subjects, glossary concepts...).
> 
> All enabled heading filters will be applied to the string.

(https://docs.moodle.org/dev/Output_functions#format_string.28.29)
So format_string() would be the wrong function for this occasion.

**Pull request merged**